packages/hathor-rpc-handler/src/rpcMethods/getBalance.ts

```diff
@@ -60,19 +61,22 @@ export async function getBalance(
 
   validateNetwork(wallet, params.network);
 
-  const balances: GetBalanceObject[] = await Promise.all(
+  const balances: GetBalanceObject[] = (await Promise.all(
     params.tokens.map(token => wallet.getBalance(token)),
-  );
+  )).flat();
 
   const prompt: GetBalanceConfirmationPrompt = {
     type: TriggerTypes.GetBalanceConfirmationPrompt,
     method: rpcRequest.method,
     data: balances
   };
 
-  const confirmed = await promptHandler(prompt, requestMetadata);
+  const confirmed = await promptHandler(
+    prompt,
+    requestMetadata
+  ) as GetBalanceConfirmationResponse;
 
-  if (!confirmed) {
+  if (!confirmed.data) {
```

Comment:
question: Should this have the additional protection of an optional chaining operator? Can `confirmed` be `null` or `undefined`?
```suggestion
  if (!confirmed?.data) {
```